### PR TITLE
Add integration test for volume controller startup.

### PR DESCRIPTION
**Pull request by jsafrane**
Tests #28002 with real etcd (unit tests have a fake one with different behavior).

@kubernetes/sig-storage

### CODE REVIEW

File: test/integration/persistent_volumes_test.go [hunk 1]
```diff
@@ -596,14 +596,121 @@ func TestPersistentVolumeMultiPVsPVCs(t *testing.T) {
...
+		pvs[i] = newPV
+	}
+
+	// Start the controller when all PVs and PVCs are already saved in etcd
+	binder.Run()
+	defer binder.Stop()
+
+	// wait for at least two sync periods for changes. No volume should be
+	// Released and no claim should be Lost during this time.
+	timer := time.NewTimer(2 * syncPeriod)
```
Comment (jingxu97) [inline test/integration/persistent_volumes_test.go]: Just wondering whether two sync periods definitely enough? (I am not very familiar with this part of code)

File: test/integration/persistent_volumes_test.go [hunk 2]
```diff
@@ -596,14 +596,121 @@ func TestPersistentVolumeMultiPVsPVCs(t *testing.T) {
...
+	defer timer.Stop()
+	finished := false
+	for !finished {
+		select {
+		case volumeEvent := <-watchPV.ResultChan():
+			volume, ok := volumeEvent.Object.(*api.PersistentVolume)
+			if !ok {
+				continue
+			}
+			if volume.Status.Phase == api.VolumeReleased {
```
Comment (jingxu97) [inline test/integration/persistent_volumes_test.go]: If the cache is not initialized properly, the Phase will change to released? Any other Phase possible?

File: test/integration/persistent_volumes_test.go (hunk 1 above)
Comment (jsafrane) [inline test/integration/persistent_volumes_test.go]: Each PV and PVC evaluated every sync period. Assuming that the PVs/PVCs are bound together, this evaluation checks that a PV still has corresponding PVC and a vice versa. When one of this pair is missing, PV gets Released or PVC gets Lost.

So, let's wait for two sync periods to make sure at least one check happened (and finished) for sure.

File: test/integration/persistent_volumes_test.go (hunk 2 above)
Comment (jsafrane) [inline test/integration/persistent_volumes_test.go]: The bug in #28002 was that some PVs could get `Released` and some PVCs `Lost`.  I reorganized the test a bit to catch also other phases as error now.

**Comment by jsafrane** [comment]
Rebased and reorganized to report *all* phase changes as error. PVs and PVCs must stay `Bound` for duration of the test.

**Comment by jsafrane** [comment]
Rebased

**Comment by jsafrane** [comment]
rebased

**Comment by jingxu97** [comment]
@ixdy, can we remove do-not-merge label? Thanks!

**Pull request merged**